woocommerce/woocommerce-ios
# My Store Analytics: Update contents for custom time range tab

Closes: #11985 
Also closes: #11980


## Description

This PR updates the content for the custom range tab on My Store analytics. Changes include:
- Updated `CustomRangeTabCreationCoordinator` logic:
  - Accept an optional start date and end date in the initializer to support updating an existing time range.
  - If start date and end date are nil, default to the last 30 days for the time range picker.
  - Check to ensure the selected time range is valid, and display an alert if the range is invalid.
- `StatsTimeRangeBarView`: Update style and enable tapping to update the custom time range. I'm using the calendar icon for the button as it's more straightforward than adding an "Edit" text like in the original design proposal.
- Updated the date formatter for the custom time range.
- Calculated the number of intervals for revenue and visit stats based on granularity for custom time range.
- Calculated refresh interval for custom time range based on interval granularity rather than time range.
- Handled updating existing custom time range.

## Testing instructions

- Enable the feature flag `customRangeInMyStoreAnalytics` and build the app.
- Log in to a test store.
- Tap the calendar icon to add a new custom range if you haven't already done that.
- Select an invalid time range (with start date >= end date) and tap Add => an alert should be displayed asking you to select again.
- Select a valid time range, a new tab for custom time range should be added.
- Confirm that the time range is a purple button with a calendar icon. Select any interval in the graph, the time should be display in gray text without the icon.
- Select the time range button to change to a different time range and select Apply. The custom range tab should be updated accordingly.
- Test with different range length to see if there's any issue with the intervals.

Please feel free to test again with the feature flag `customRangeInMyStoreAnalytics` off to confirm that everything works as previously.

## Screenshots

Invalid time range | Valid time range
--- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/a0a17fdc-d580-46a6-8721-ae644a62ab94" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f7aec315-ed58-4e16-b089-5ccbfb6a0ff2" width=320 />


---
